lib/Config.ts

```diff
@@ -59,6 +59,8 @@ class Config {
 
     this.p2p = {
       listen: true,
+      discover: false,
+      detectExternalIp: true,
```

Comment:
I think the default should be false. In most cases, this will run behind NAT and the detection will fail. Better to have this as an advanced option.

Reply:
I would gladly make changes after conclusion @sangaman ?

Reply:
That would change the original intention of having this feature, I would prefer to keep it that way. If it fails, it's not a problem, it's no different than if we didn't try in the first place. Fwiw, personally I am behind NAT and it detects the external IP for my router fine.